comp/forwarder/eventplatform/eventplatformimpl/epforwarder.go

```diff
@@ -344,20 +344,38 @@ func Diagnose() []diagnose.Diagnosis {
 				Diagnosis: fmt.Sprintf("Connectivity to `%s` is Ok", url),
 			})
 		} else {
-			diagnoses = append(diagnoses, diagnose.Diagnosis{
+			diag := diagnose.Diagnosis{
 				Status:      diagnose.DiagnosisFail,
 				Category:    desc.category,
 				Name:        name,
 				Diagnosis:   fmt.Sprintf("Connection to `%s` failed", url),
 				Remediation: "Please validate Agent configuration and firewall to access " + url,
 				RawError:    err.Error(),
-			})
+			}
+			diag = maybeTestWithPQDN(endpoints.Main, diag)
+			diagnoses = append(diagnoses, diag)
 		}
 	}
 
 	return diagnoses
 }
 
+// Detect if the connection failed because of using a FQDN by trying with a PQDN
+func maybeTestWithPQDN(endpoint config.Endpoint, diag diagnose.Diagnosis) diagnose.Diagnosis {
+	host := endpoint.Host
+
+	if strings.HasSuffix(host, ".") {
+		log.Infof("The connection to %s with a FQDN failed; attempting with a PQDN", host)
+
+		endpoint.Host = strings.TrimSuffix(host, ".")
+		_, err := logshttp.CheckConnectivityDiagnose(endpoint, pkgconfigsetup.Datadog())
+		if err == nil {
+			diag.Remediation = "The connection with a FQDN failed but is possible with PQDN, please check firewall and/or proxy configuration"
```

Comment:
The remediation is not very clear or actionable, I think it would be best to make it explicit what FQDN/PQDN are (very likely not clear for most users) by giving an example, or event the actual value used by the agent, and mentioning setting `convert_dd_site_fqdn.enabled: false` to make the agent stop using the FQDN

Reply:
Good point, implemented!